c7n/resources/opsworks.py

```diff
@@ -155,6 +207,106 @@ def process_stack(self, stack):
             self.log.exception(
                 "Exception stopping stack:\n %s" % e)
 
+@OpsworkStack.action_registry.register('start')
+class StartStack(BaseAction):
+    """Action to start Opswork Stack
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: opswork-stop
+                resource: opswork-stack
+                actions:
+                  - start
+    """
+
+    schema = type_schema('start')
+    permissions = ("opsworks:StartStack",)
+
+    def process(self, stacks):
+        with self.executor_factory(max_workers=10) as w:
+            list(w.map(self.process_stack, stacks))
+
+    def process_stack(self, stack):
+        client = local_session(
+            self.manager.session_factory).client('opsworks')
+        try:
+            stack_id = stack['StackId']
+            client.start_stack(StackId=stack_id)
+        except ClientError as e:
+            self.log.exception(
+                "Exception starting stack:\n %s" % e)
+
+@OpsworkStack.action_registry.register('tag')
+@OpsworkStack.action_registry.register('mark')
+class Tag(tags.Tag):
+    """Action to add a tag to an OpsWorks stack
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: opswork-stack-add-owner-tag
+                resource: opswork-stack
+                filters:
+                  - "tag:OwnerName": absent
+                actions:
+                  - type: tag
+                    key: OwnerName
+                    value: OwnerName
+    """
+    schema = type_schema(
+        'tag',
+        key={'type': 'string'},
+        value={'type': 'string'},
+        aliases=('mark',)
+    )
+
+    permissions = ('opsworks:TagResource',)
+    batch_size = 1
+
+    def process_resource_set(self, stacks, tags):
+        client = local_session(
+            self.manager.session_factory
+        ).client('opsworks')
+        tag_dict = {}
+        for t in tags:
+           tag_dict[t['Key']] = t['Value']
+
+        for stack in stacks:
+            client.tag_resource(ResourceArn=stack['Arn'], Tags=tag_dict)
+
+@OpsworkStack.action_registry.register('remove-tag')
+@OpsworkStack.action_registry.register('untag')
+@OpsworkStack.action_registry.register('unmark')
+class RemoveTag(tags.RemoveTag):
+    """Action to remove tag(s) from OpsWorks stack
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: opswork-remove-tag
+                resource: opswork-stack
+                filters:
+                  - "tag:OutdatedTag": present
+                actions:
+                  - type: remove-tag
+                    tags: ["OutdatedTag"]
+    """
+    concurrency = 2
```

Comment:
same thing, concurrency and batch_size are both unused right now so they should be removed

Reply:
fixed